src/types/ServiceInfo.ts

```diff
@@ -362,42 +362,24 @@ export interface ServiceDetailsInfo {
   labels?: { [key: string]: string };
   type: string;
   name: string;
-  created_at: string;
-  resource_version: string;
+  createdAt: string;
+  resourceVersion: string;
   ip: string;
   ports?: Port[];
   endpoints?: Endpoints[];
-  istio_sidecar: boolean;
+  istioSidecar: boolean;
   pods?: Pod[];
   deployments?: Deployment[];
   routeRules?: RouteRule[];
   destinationPolicies?: DestinationPolicy[];
   virtualServices?: VirtualService[];
   destinationRules?: DestinationRule[];
-  dependencies?: Map<string, string[]>;
+  dependencies?: { [key: string]: string[] };
   health?: Health;
 }
 
-// Temporary interface due to tiny differences between the model from REST API and the ServiceDetailsInfo used above.
-//  They should be cleaned & merged asap
-export interface ServiceDetailsInfoFromAPI {
-  labels?: { [key: string]: string };
-  type: string;
-  name: string;
-  created_at: string;
-  resource_version: string;
-  ip: string;
-  ports?: Port[];
-  endpoints?: Endpoints[];
-  pods?: Pod[];
-  deployments?: Deployment[];
-  route_rules?: RouteRule[];
-  destination_policies?: DestinationPolicy[];
-  virtual_services?: VirtualService[];
-  destination_rules?: DestinationRule[];
-  dependencies?: Map<string, string[]>;
-  health?: Health;
-}
+// validations are grouped per 'objectType' first in the first map and 'name' in the inner map
+export type Validations = { [k1: string]: { [k2: string]: ObjectValidation } };
```

Comment:
k1 -> objectType
k2 -> name
I guess that in case of multiple maps, it helps to get the scenario.
But as it is commented out, at least I would use key1, key2.